### PR TITLE
fix style guide violations

### LINKED ISSUE #2: Fix style guide violations
When reviewing the initial commit of code that made its way over from the `ponyc` repo, @Theodus found some style guide violations that should be fixed. See the original PR for more details: https://github.com/ponylang/net-ssl/pull/1

### PR DESCRIPTION
**Pull request by Theodus**
resolve #2

### CODE REVIEW

File: examples/simple-example/simple-example.pony [hunk 1]
```diff
@@ -49,7 +47,9 @@ class Listener is TCPListenNotify
   var _service: String = ""
   var _count: USize = 0
 
-  new create(sslctx: SSLContext, auth: AmbientAuth, out: OutStream, limit: USize) =>
+  new iso create(
+    sslctx: SSLContext, auth: AmbientAuth, out: OutStream, limit: USize)
```
Comment (SeanTAllen) [inline examples/simple-example/simple-example.pony]: i don't believe this is correct by the style guide.

shouldn't it be:

```
new iso create(sslctx: SSLContext, auth: AmbientAuth, out: OutStream, 
  limit: USize)
=>
```

File: examples/simple-example/simple-example.pony [hunk 2]
```diff
@@ -93,8 +93,11 @@ class Listener is TCPListenNotify
 
     try
       _out.print("client starting")
-      let ssl = _sslctx.client()?
-      TCPConnection(_auth, SSLConnection(ClientSide(_out), consume ssl), _host, _service)
+      TCPConnection(
+        _auth,
+        SSLConnection(ClientSide(_out), _sslctx.client()?),
+        _host,
+        _service)
```
Comment (SeanTAllen) [inline examples/simple-example/simple-example.pony]: is this correct by the style guide? i thought it was supposed to only wrap extra things, not do 1 per line.

File: examples/simple-example/simple-example.pony [hunk 3]
```diff
@@ -113,8 +116,8 @@ class ServerSide is TCPConnectionNotify
       conn.write("server says hi")
     end
 
-  fun ref received(conn: TCPConnection ref, data: Array[U8] iso,
```
Comment (SeanTAllen) [inline examples/simple-example/simple-example.pony]: I thought that what you are replacing is correct according to the style guide.

File: examples/simple-example/simple-example.pony [hunk 4]
```diff
@@ -147,11 +150,11 @@ class ClientSide is TCPConnectionNotify
   fun ref connect_failed(conn: TCPConnection ref) =>
     _out.print("connect failed")
 
-  fun ref received(conn: TCPConnection ref, data: Array[U8] iso,
-    times: USize): Bool
+  fun ref received(
+    conn: TCPConnection ref, data: Array[U8] iso, times: USize): Bool
```
Comment (SeanTAllen) [inline examples/simple-example/simple-example.pony]: neither of these seems to match the style guide. i think we have a gap in the style guide.

File: examples/simple-example/simple-example.pony [hunk 5]
```diff
@@ -49,7 +47,9 @@ class Listener is TCPListenNotify
   var _service: String = ""
   var _count: USize = 0
 
-  new create(sslctx: SSLContext, auth: AmbientAuth, out: OutStream, limit: USize) =>
+  new iso create(
+    sslctx: SSLContext, auth: AmbientAuth, out: OutStream, limit: USize)
+  =>
```
Comment (SeanTAllen) [inline examples/simple-example/simple-example.pony]: from my reading of the style guide, these should be 1 per line.

File: examples/simple-example/simple-example.pony (hunk 4 above)
Comment (Theodus) [reply examples/simple-example/simple-example.pony]: I agree, the style guide is a bit ambiguous on this point.

File: examples/simple-example/simple-example.pony [hunk 6]
```diff
@@ -6,39 +6,37 @@ use "../../net_ssl"
...
-            SSLContext
-              .>set_authority(FilePath(auth, "../../examples/simple-example/cert.pem")?)?
-              .>set_cert(
-                FilePath(auth, "../../examples/simple-example/cert.pem")?,
-                FilePath(auth, "../../examples/simple-example/key.pem")?)?
-              .>set_client_verify(true)
-              .>set_server_verify(true)
-          end
+    let auth =
+      try env.root as AmbientAuth
```
Comment (SeanTAllen) [inline examples/simple-example/simple-example.pony]: this isn't covered in the style guide but i think the try should be on its own line here.

i think we should make that clear in the style guide. if you have a multiline try, the the try should be on its own line.

File: net_ssl/alpn.pony [hunk 7]
```diff
@@ -1,7 +1,8 @@
 use "net"
 
 interface ALPNProtocolNotify
-  fun ref alpn_negotiated(conn: TCPConnection ref, protocol: (String val | None)): None
+  fun ref alpn_negotiated(
+    conn: TCPConnection ref, protocol: (String | None)): None
```
Comment (SeanTAllen) [inline net_ssl/alpn.pony]: i think this is incorrect according to the style guide

File: net_ssl/ssl_context.pony [hunk 8]
```diff
@@ -239,12 +240,13 @@ class val SSLContext
     Requires OpenSSL >= 1.0.2
     """
     ifdef "openssl_1.1.x" then
-      @SSL_CTX_set_alpn_select_cb[None](_ctx, addressof SSLContext._alpn_select_cb, resolver)
+      @SSL_CTX_set_alpn_select_cb[None](
+        _ctx, addressof SSLContext._alpn_select_cb, resolver)
```
Comment (SeanTAllen) [inline net_ssl/ssl_context.pony]: i believe this incorrect according to the style guide.

File: net_ssl/ssl_context.pony [hunk 9]
```diff
@@ -256,28 +258,30 @@ class val SSLContext
     ifdef "openssl_1.1.x" then
       try
         let proto_list = _ALPNProtocolList.from_array(protocols)?
-        let result = @SSL_CTX_set_alpn_protos[I32](_ctx, proto_list.cpointer(), proto_list.size())
+        let result =
+          @SSL_CTX_set_alpn_protos[I32](
+            _ctx, proto_list.cpointer(), proto_list.size())
```
Comment (SeanTAllen) [inline net_ssl/ssl_context.pony]: i believe this incorrect according to the style guide.

File: net_ssl/x509.pony [hunk 10]
```diff
@@ -22,26 +22,20 @@ primitive X509
...
-    if len < 0 then
-      error
-    end
+    if len < 0 then error end
 
     let common = recover String(len.usize()) end
-    @X509_NAME_get_text_by_NID[I32](subject, I32(13), common.cstring(),
-      len + 1)
+    @X509_NAME_get_text_by_NID[I32](
+      subject, I32(13), common.cstring(), len + 1)
```
Comment (SeanTAllen) [inline net_ssl/x509.pony]: i believe this incorrect according to the style guide.

File: examples/simple-example/simple-example.pony (hunk 6 above)
Comment (Theodus) [reply examples/simple-example/simple-example.pony]: Do you mean that the `env.root as AmbientAuth` bit should be on a separate line after the `try` keyword?

File: net_ssl/x509.pony [hunk 11]
```diff
@@ -62,8 +56,8 @@ primitive X509
     end
 
     let stack =
-      @X509_get_ext_d2i[Pointer[_GeneralNameStack]](cert, I32(85),
-        Pointer[U8], Pointer[U8])
+      @X509_get_ext_d2i[Pointer[_GeneralNameStack]](
+        cert, I32(85), Pointer[U8], Pointer[U8])
```
Comment (SeanTAllen) [inline net_ssl/x509.pony]: i believe this incorrect according to the style guide.

**Comment by SeanTAllen** [comment]
@Theodus I left some comments on where i think some changes are still against the style guide.

**Comment by SeanTAllen** [comment]
@Theodus we don't have any style guide about FFI calls. I think we should add a style guide item that all FFI functions are declared ahead of time with a `use`. Thoughts?

File: examples/simple-example/simple-example.pony (hunk 4 above)
Comment (SeanTAllen) [reply examples/simple-example/simple-example.pony]: As I read the style guide, it should be 1 parameter per line as this is a multiline method signature.

File: examples/simple-example/simple-example.pony (hunk 6 above)
Comment (SeanTAllen) [reply examples/simple-example/simple-example.pony]: Yes.

**Comment by SeanTAllen** [comment]
@Theodus looks good to me. when it passes CI, you can squash and merge.

**Comment by Theodus** [comment]
@SeanTAllen yup, will do

**Pull request merged**